### PR TITLE
Create parcel

### PR DESCRIPTION
**Pull request by dantleech**
This PR adds the Create Parcel operatoin and simplifies the infrastructure to make adding future methods simpler.

- Renamed LineItemUpdaterContext => ApplicationContext
- Removes knowledge of operations from the HttpClient (it now just has METHOD, ENDPOINT and OBJECT as parameters).
- Introduces a generic "request agent" responsible for instnatiating the request objects from arrays and sending them to the API.

NOTE: This PR implies refactoring the Order Export to work in the same way as the new methods.

### CODE REVIEW

File: features/bootstrap/Contexts/OrderExportContext.php [hunk 1]
```diff
@@ -131,6 +131,9 @@ public function orderIsExported(string $orderId)
      */
     public function theExportForOrderShouldBeSuccessful()
     {
-        Assert::true($this->testApplicationProxy->isLastResponseSuccessful(), $this->testApplicationProxy->getLastErrorMessage());
+        Assert::true(
+            $this->testApplicationProxy->isLastResponseSuccessful(),
+            $this->testApplicationProxy->getLastResponse()->getErrorMessage()
```
Comment (dantleech) [inline features/bootstrap/Contexts/OrderExportContext.php]: This was showing the last exception message, not the last API message

File: features/bootstrap/Services/TestApplicationProxy.php [hunk 2]
```diff
@@ -48,6 +48,20 @@ public function exportOrder(array $orderParameters)
         }
     }
 
+    public function createParcel(array $parameters): void
+    {
+        try {
+            $this->lastApiResponse = $this->application->createParcel($parameters);
+        } catch (OneStockException $e) {
+            $this->lastApiException = $e;
```
Comment (dantleech) [inline features/bootstrap/Services/TestApplicationProxy.php]: Could remove this class completely ....

File: src/Inviqa/OneStock/RequestAgent.php [hunk 3]
```diff
@@ -31,6 +32,16 @@ public function update(array $lineItemUpdateParameters): OneStockResponse
             'items' => $lineItemUpdateParameters,
         ]);
 
-        return $this->apiClient->updateLineItems($request);
+        return $this->apiClient->request('PATCH', 'multi/line_items', $request);
+    }
```
Comment (dantleech) [inline src/Inviqa/OneStock/RequestAgent.php]: rename "update" above to "lineItemUpdate"

**Comment by dantleech** [comment]
> All looks good, but when will the order export refactored?

:thinking:

File: features/bootstrap/Services/TestApplicationProxy.php (hunk 2 above)
Comment (ubick) [reply features/bootstrap/Services/TestApplicationProxy.php]: Good idea :+1:

File: src/Inviqa/OneStock/RequestAgent.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Inviqa\OneStock;
+
+use DTL\Invoke\Invoke;
+use Inviqa\OneStock\Client\ApiClient;
+use Inviqa\OneStock\LineUpdater\LineItemUpdateRequest;
+use Inviqa\OneStock\Parcel\ParcelCreateRequest;
+
+final class RequestAgent
```
Comment (ubick) [inline src/Inviqa/OneStock/RequestAgent.php]: Only publicly available classes should live in the root OneStock namespace.

Should the `RequestAgent` be better placed in some kind of `Request` namespace? I would imagine other services could live there as well.

**Pull request merged**